src/lava/proc/lif/process.py

```diff
@@ -7,7 +7,29 @@
 from lava.magma.core.process.ports.ports import InPort, OutPort
 
 
-class LIF(AbstractProcess):
+class AbstractLIF(AbstractProcess):
+    """Abstract class for variables common to all neurons with leaky
+    integrator dynamics."""
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        shape = kwargs.get("shape", (1,))
+        du = kwargs.pop("du", 0)
+        dv = kwargs.pop("dv", 0)
+        bias = kwargs.pop("bias", 0)
+        bias_exp = kwargs.pop("bias_exp", 0)
+
+        self.shape = shape
+        self.a_in = InPort(shape=shape)
+        self.s_out = OutPort(shape=shape)
+        self.u = Var(shape=shape, init=0)
+        self.v = Var(shape=shape, init=0)
+        self.du = Var(shape=(1,), init=du)
+        self.dv = Var(shape=(1,), init=dv)
```

Comment:
du and dv are part of compartment profile in Loihi-1. There are only 32 of those possible. This means, 32 distinct populations of neurons can have their own du and dv. So if one wants to assign different du and dv values, one can instantiate a new LIF Process with those values.. from the perspective of compiling to the hardware, this will result in using a different compartment profile for each population.

Reply:
I'm aware of the hardware constraint, but I don't think we should constraint the software to a very specific (version) of the hardware. Because Lava is intended as a more general framework, we could offer total flexibility in the more abstract processes/models and then constrain the more specialized ones. If we put constraints on the high-level abstractions then we may reduce their usability.